myriad-scheduler/src/test/java/org/apache/myriad/scheduler/fgs/YarnNodeCapacityManagerSpec.groovy

```diff
@@ -115,7 +117,7 @@ class YarnNodeCapacityManagerSpec extends FGSTestBaseSpec {
         then:
         zeroNM.getTotalCapability().getMemory() == 2048
         zeroNM.getTotalCapability().getVirtualCores() == 2
-        1 * rmContext.getDispatcher().getEventHandler().handle(_ as NodeResourceUpdateSchedulerEvent)
+        1 * yarnScheduler.updateNodeResource( _ as RMNode, _ as ResourceOption)
```

Comment:
The reason I bring it up is that I believe that unless this test is synchronized we won't be testing the exact behavior of the method. Namely, the way the synchronized method is implemented above places a lock on the yarnScheduler object preventing other methods from being called on the object anywhere else in the class. Since by default groovy tests run in an asynchronous manner the behavior of the test may be different if the object is used anywhere else within the test class.